guvnor-ala/guvnor-ala-services-api/src/main/java/org/guvnor/ala/services/api/RuntimeProvisioningService.java

```diff
@@ -0,0 +1,159 @@
+            throws BusinessException;
+
+    /*
+     * Get all the registered Providers
+     * @return ProviderList containing all the registered providers
+     * @see ProviderList
+     * @throw BusinessException in case of an internal exception
+     */
+    @GET
+    @Produces( value = APPLICATION_JSON )
+    @Path( "providers" )
+    ProviderList getProviders( @QueryParam( "page" ) @DefaultValue( "0" ) Integer page,
+            @QueryParam( "pageSize" ) @DefaultValue( "10" ) Integer pageSize,
+            @QueryParam( "sort" ) String sort, @QueryParam( "sortOrder" ) @DefaultValue( "true" ) boolean sortOrder )
+            throws BusinessException;
+
+    /*
+     * Register a new Provider
+     * @param ProviderConfig used to create the new Provider
+     * @throw BusinessException in case of an internal exception
+     */
+    @POST
+    @Consumes( value = APPLICATION_JSON )
+    @Path( "providers" )
+    void registerProvider( @NotNull ProviderConfig conf ) throws BusinessException;
+
+    /*
+     * Unregister an existing Provider
+     * @param String provider name
+     * @throw BusinessException in case of an internal exception
+     */
+    @DELETE
+    @Path( "providers" )
+    void unregisterProvider( @FormParam( value = "name" ) String name ) throws BusinessException;
+
+    /*
+     * Create a new Runtime
+     * @param RuntimeConfig containing the configuration used to create the new Runtime
+     * @throw BusinessException in case of an internal exception
+     */
+    @POST
+    @Path( "runtimes/" )
+    @Consumes( value = APPLICATION_JSON )
+    @Produces( value = APPLICATION_JSON )
+    String newRuntime( @NotNull RuntimeConfig conf ) throws BusinessException;
+
+    /*
+     * Destroy an existing  Runtime
+     * @param String runtimeId of the runtime to destroy
+     * @throw BusinessException in case of an internal exception
+     */
+    @DELETE
+    @Consumes( value = APPLICATION_JSON )
+    @Path( "runtimes/{id}" )
+    void destroyRuntime( @PathParam( value = "id" ) String runtimeId ) throws BusinessException;
+
+    /*
+     * Get All Runtimes
+     * @return RuntimeList containing all the registered Runtimes 
+     * @throw BusinessException in case of an internal exception
+     */
+    @GET
+    @Produces( value = APPLICATION_JSON )
+    @Path( "runtimes/" )
+    RuntimeList getRuntimes( @QueryParam( "page" ) @DefaultValue( "0" ) Integer page,
+            @QueryParam( "pageSize" ) @DefaultValue( "10" ) Integer pageSize,
+            @QueryParam( "sort" ) String sort, @QueryParam( "sortOrder" ) @DefaultValue( "true" ) boolean sortOrder )
+            throws BusinessException;
+
+    /*
+     * Start a given Runtime
+     * @param String runtimeId of the runtime to be started
+     * @throw BusinessException in case of an internal exception
+     */
+    @DELETE
+    @Path( "runtimes/{id}/state" )
+    void startRuntime( @PathParam( value = "id" ) String runtimeId ) throws BusinessException;
```

Comment:
maybe you could use instance instead of state... though that's a matter of preference